src/ui/public/courier/data_source/__tests__/_migrate_filter.js

```diff
@@ -0,0 +1,44 @@
+import expect from 'expect.js';
+import _ from 'lodash';
+import { migrateFilter } from '../_migrate_filter';
+
+describe('migrateFilter', function () {
+
+  const oldMatchPhraseFilter = {
+    match: {
+      fieldFoo: {
+        query: 'foobar',
+        type: 'phrase'
+      }
+    }
+  };
+
+  const newMatchPhraseFilter = {
+    match_phrase: {
+      fieldFoo: {
+        query: 'foobar'
+      }
+    }
+  };
+
+  // https://github.com/elastic/elasticsearch/pull/17508
+  it('should migrate match filters of type phrase', function () {
+    const migratedFilter = migrateFilter(oldMatchPhraseFilter);
+    expect(_.isEqual(migratedFilter, newMatchPhraseFilter)).to.be(true);
+  });
+
+  it('should not modify the original filter', function () {
+    const oldMatchPhraseFilterCopy = _.clone(oldMatchPhraseFilter, true);
+    migrateFilter(oldMatchPhraseFilter);
+    expect(_.isEqual(oldMatchPhraseFilter, oldMatchPhraseFilterCopy)).to.be(true);
+  });
+
+  it('should return the original filter if no migration is necessary', function () {
+    const originalFilter = {
+      match_all: {}
+    };
+    const migratedFilter = migrateFilter(originalFilter);
+    expect(migratedFilter).to.be(originalFilter);
```

Comment:
I talking more about the idea of checking the equality between `migratedFilter` and `originalFilter` instead of checking that `migratedFilter` is not eql to `newMatchPhraseFilter`. It seems that the former would catch more errors, but I wasn't sure if I was missing something about what you were suggesting.

As an aside, I agree that the error reporting sucks when using `_.isEqual` but the problem I have with expect.js's `eql` function is that it uses loose equality (`==`) so there are subtle bugs it would miss. Another reason I wish we were using Chai.